server/src/test/java/io/crate/planner/ExplainPlannerTest.java

```diff
@@ -305,6 +305,13 @@ public void test_explain_verbose_costs_adds_estimated_rows_to_output() throws Ex
                   └ Join[INNER | (x = x)] (rows=0)
                     ├ Collect[doc.a | [x] | true] (rows=100)
                     └ Collect[doc.b | [x] | true] (rows=100)"""},
+            new Object[]{
+                "optimizer_move_constant_join_conditions_beneath_join",
```

Comment:
The description mentions that this change would reduce the steps, but here is now an extra step. Is this correct?

What is an example where steps are reduced?

Reply:
This step here is only marking, but not changing the plan. `MoveFilterBeneathJoin` matched before on `AbstractJoinPlan` but returned `JoinPlan`. Joins are initialized as `JoinPlan` and will be converted to `NestedLoop` or `HashJoin` eventually. `MoveFilterBeneathJoin` can then match again because `NestedLoop` is also an `AbstractJoinPlan` and return `JoinPlan` again. I had a few queries where this was happening.

Reply:
Would you prefer a real-world example query?